src/test/scala/org/broadinstitute/dsde/workbench/sam/api/ManagedGroupRoutesSpec.scala

```diff
@@ -0,0 +1,97 @@
+package org.broadinstitute.dsde.workbench.sam.api
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import org.broadinstitute.dsde.workbench.model.{UserInfo, WorkbenchEmail, WorkbenchUserId}
+import org.broadinstitute.dsde.workbench.sam.{TestSupport, model}
+import org.broadinstitute.dsde.workbench.sam.model._
+import org.broadinstitute.dsde.workbench.sam.service.ManagedGroupService
+import org.scalatest.{BeforeAndAfter, FlatSpec, GivenWhenThen, Matchers}
+
+/**
+  * Created by dvoet on 6/7/17.
+  */
+class ManagedGroupRoutesSpec extends FlatSpec with Matchers with ScalatestRouteTest with TestSupport with BeforeAndAfter {
+
+  private val ownerRoleName = ResourceRoleName("admin")
+  private val ownerPolicyName = AccessPolicyName(ownerRoleName.value)
+  private val memberPolicyName = AccessPolicyName(ManagedGroupService.MemberRoleName.value)
+  private val accessPolicyNames = Set(ownerPolicyName, memberPolicyName)
+  private val policyActions: Set[ResourceAction] = accessPolicyNames.flatMap(policyName => Set(SamResourceActions.sharePolicy(policyName), SamResourceActions.readPolicy(policyName)))
+  private val resourceActions = Set(ResourceAction("delete")) union policyActions
+  private val resourceActionPatterns = resourceActions.map(action => ResourceActionPattern(action.value))
+  private val defaultOwnerRole = ResourceRole(ownerRoleName, resourceActions)
+  private val defaultRoles = Set(defaultOwnerRole, ResourceRole(ManagedGroupService.MemberRoleName, Set.empty))
+  private val managedGroupResourceType = ResourceType(ManagedGroupService.ManagedGroupTypeName, resourceActionPatterns, defaultRoles, ownerRoleName)
+  private val resourceTypes = Map(managedGroupResourceType.name -> managedGroupResourceType)
+  private val groupId = "foo"
+
+  def assertGroupDoesNotExist(samRoutes: TestSamRoutes, groupId: String = groupId) {
+    Get(s"/api/group/$groupId") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.NotFound
+    }
+  }
+
+  def assertCreateGroup(samRoutes: TestSamRoutes, groupId: String = groupId): Unit = {
+    Post(s"/api/group/$groupId") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.NoContent
+      responseAs[String].isEmpty shouldEqual true
+    }
+  }
+
+  def assertGetGroup(samRoutes: TestSamRoutes, groupId: String = groupId) = {
+    Get(s"/api/group/$groupId") ~> samRoutes.route ~> check {
+      val expectedResource = Resource(ManagedGroupService.ManagedGroupTypeName, ResourceId(groupId))
+      status shouldEqual StatusCodes.OK
+      responseAs[String] should include (s"${groupId}@${samRoutes.resourceService.emailDomain}")
```

Comment:
In general, route tests should test return codes and input/output formats but not response values. The service level tests should test that output is actually as expected.

Reply:
fixed